src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/BlockSyntaxExtensions.cs

```diff
@@ -62,6 +62,17 @@ internal static class BlockSyntaxExtensions
                 block.TryConvertToExpressionBody(languageVersion, preference, out var expression, out semicolonToken))
             {
                 arrowExpression = SyntaxFactory.ArrowExpressionClause(expression);
+
+                if (block.GetRequiredParent().Kind() == SyntaxKind.GetAccessorDeclaration)
```

Comment:
small nit (can take or leave it) - can extract out a variable for block.GetRequiredParent() and re-use that here and below